jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/AnyUserLoginService.java

```diff
@@ -0,0 +1,70 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe;
+
+import java.util.function.Function;
+import javax.security.auth.Subject;
+
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.UserIdentity;
+import org.eclipse.jetty.security.UserPrincipal;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Session;
+
+public class AnyUserLoginService implements LoginService
```

Comment:
Javadocs, as I think `ANY_USER` is not any user.
What's the point of this class, if it does not do any login?
Can you make it internal?

Reply:
The point of this is that if you want to allow authenticate any user, not just a set of users known by the `LoginService`.

I would say this is the most common use case for Sign-In with Ethereum.

For example other `LoginService` impls which could be used would be `HashLoginService` or `JDBCLoginService`, but these would not be able to log in someone not already in the database/hashmap.

But maybe this also needs to be able to delegate to another LoginService to get roles. In this case it would need to be public not internal.